### PR TITLE
[Improvement-5773][server] need to support two parameters related to task

### LINKED ISSUE #5773: [Improvement][server] need to support two parameters related to task
**Describe the question**
When I'm using the shell task ,I need the instance id of task and the absolute path of task.

**What are the current deficiencies and the benefits of improvement**

**Which version of DolphinScheduler:**
 -[dev]

**Describe alternatives you've considered**

### PR DESCRIPTION
**Pull request by calvinjiang**
## Purpose of the pull request
this pr close #5773 

When I'm using the shell task ,I need the instance id of task and the absolute path of task.

## Verify this pull request

This change added tests and can be verified as follows:

org.apache.dolphinscheduler.server.utils.ParamUtilsTest.testConvertForParamsRelatedTask

### CODE REVIEW

**Review by zhuangchong — APPROVED** [review]
LGTM

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/ParamUtils.java [hunk 1]
```diff
@@ -95,6 +101,67 @@
...
+
+        if (globalParamsMap != null) {
+            timeParams.putAll(globalParamsMap);
+        }
+
+        if (Strings.isNotBlank(taskExecutionContext.getExecutePath())) {
+            timeParams.put(Constants.PARAMETER_TASK_EXECUTE_PATH,taskExecutionContext.getExecutePath());
+        }
+        timeParams.put(Constants.PARAMETER_TASK_INSTANCE_ID,Integer.toString(taskExecutionContext.getTaskInstanceId()));
+
```
Comment (CalvinKirs) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/ParamUtils.java]: 1:There is an ambiguity in using timesParam to store temporary variables. timesParam mainly stores parameters of time type.

2:Is the result correct? Simply looking at the code, it does not seem to complete the final replacement.

3:If we want to support it, we suggest that other task types adopt the same approach to be consistent. (Maybe other tasks have the same need)

Comment (calvinjiang) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/ParamUtils.java]: 1：For sure ,it's not pretty good way to use timesParam to store temporary variables, I will rename timesParam.

2: The result is correct . This feature has passed the unit test, and we have validated it in our environment. 
image

3：Sounds good. We will do that in next pr.

Comment (CalvinKirs) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/ParamUtils.java]: It is best to create an issue to track this issue.

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Comment by CalvinKirs** [comment]
Deeply thanks for your contribution, this is a good beginning, considering that it's your first contribution, I think we can get deep communication, you can contact me through mail or add WeChat(Kris_Evil), when mail or added, please tell me who you are, I think I can help to familiar with the DolphinScheduler if you meet with problems.

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/ParamUtils.java (hunk 1 above)
Comment (CalvinKirs) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/ParamUtils.java]: hi,how is it getting?

Comment (calvinjiang) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/ParamUtils.java]: > hi,how is it getting?

First of all: create a new shell task, and then add this two parameters in the field of customer defined parameter. You can use these self-designed parameters in the script.

image

Second: save this task and run this flow to check this result.

image

Comment (CalvinKirs) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/ParamUtils.java]: I am referring to legacy question.：）

Comment (calvinjiang) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/ParamUtils.java]: > I am referring to legacy question.：）

Got it. I've made a new issue for that. I'll implement this function to support the rest of type of tasks ASAP.

New issue is :https://github.com/apache/dolphinscheduler/issues/5852

**Pull request merged**